junit-platform-commons/src/main/java/org/junit/platform/commons/support/ReflectionSupport.java

```diff
@@ -137,6 +157,30 @@ public static List<Class<?>> findAllClassesInClasspathRoot(URI root, Predicate<C
 		return ReflectionUtils.findAllClassesInClasspathRoot(root, classFilter, classNameFilter);
 	}
 
+	/**
+	 * Find all {@linkplain Resource resources} in the supplied classpath {@code root}
+	 * that match the specified {@code resourceFilter} and {@code resourceNameFilter}
+	 * predicates.
+	 *
+	 * <p>The classpath scanning algorithm searches recursively in subpackages
+	 * beginning with the root of the classpath.
+	 *
+	 * @param root the URI for the classpath root in which to scan; never
+	 * {@code null}
+	 * @param resourceFilter the resource type filter; never {@code null}
+	 * @param resourceNameFilter the resource name filter; never {@code null}
+	 * @return an immutable list of all such resources found; never {@code null}
+	 * but potentially empty
+	 * @see #findAllResourcesInPackage(String, Predicate, Predicate)
+	 * @see #findAllResourcesInModule(String, Predicate, Predicate)
+	 */
+	@API(status = EXPERIMENTAL, since = "1.11")
+	public static List<Resource> findAllResourcesInClasspathRoot(URI root, Predicate<Resource> resourceFilter,
+			Predicate<String> resourceNameFilter) {
```

Comment:
:question: Why are the variants with two `Predicate` parameters back?

Reply:
Because of https://github.com/junit-team/junit5/pull/3718#discussion_r1639617633. To ensure resources are found using the same semantics as `ClassLoader.getResource`, the best solution is to use `ClassLoader.getResource`. But the class loader is slow, so to prevent calling the classloader unnecessarily we end up with the same solution as the `findAllClasses*` variants. 

I've put an illustrative example of the expected behavior for shadowed resources in https://github.com/junit-team/junit5/pull/3705/commits/794793a9f396f60acd5f451e8305deecbd71b062#diff-67ad34a98151fe1159d3dbaa3b89cf23f691b6a8eac64231424ee48bd72c14d2R234 .